server/webapp/dbmodels.py

```diff
@@ -674,3 +682,65 @@ def recursive_delete(self, synchronize_session=False):
 
     def as_file(self, loaddir, filename=None):
         return db_model_as_file(self, loaddir, filename, 'name', 'prg')
+
+
+@swagger.model
+class ScenariosDb(db.Model):
+
+    __tablename__ = 'scenarios'
+
+    resource_fields = {
+        'id': Uuid,
+        'progset_id': Uuid,
+        'scenario_type': fields.String,
+        'active': fields.Boolean,
+        'name': fields.String,
+        'parset_id': Uuid,
+        'pars': Json(attribute='pars')
+    }
+
+    id = db.Column(UUID(True), server_default=text("uuid_generate_v1mc()"), primary_key=True)
+    project_id = db.Column(UUID(True), db.ForeignKey('projects.id'))
+    name = db.Column(db.String)
+    scenario_type = db.Column(db.String)
+    active = db.Column(db.Boolean)
+    progset_id = db.Column(UUID(True), db.ForeignKey('progsets.id'))
+    parset_id = db.Column(UUID(True), db.ForeignKey('parsets.id'))
+    blob = db.Column(JSON)
+
+    def __init__(self, project_id, parset_id, name, scenario_type,
+                 active=False, progset_id=None, blob={}):
+
+        self.project_id = project_id
+        self.name = name
+        self.scenario_type = scenario_type
+        self.active = active
+        self.progset_id = progset_id
+        self.parset_id = parset_id
+        self.blob = blob
+
+    @property
+    def pars(self):
+        print(self.blob)
+        return self.blob.get('pars', [])
+
+    def hydrate(self):
+
+        from server.webapp.utils import load_progset, load_parset
+
+        parset = load_parset(self.project_id, self.parset_id)
+
+        if self.scenario_type == "Program":
+
+            progset = utils.load_progset(self.project_id, self.progset_id)
+
+            return op.Progscen(name=self.name,
+                               parsetname=parset.name,
+                               progsetname=progset.name,
+                               **self.blob)
+
+        elif self.scenario_type == "Parameter":
+
+            return op.Parscen(name=self.name,
+                              parsetname=parset.name,
+                              **self.blob)
```

Comment:
Hydrate ok, but I don't see any "restore" code. (it can be added as a separate card)